test/model/content_test.dart

```diff
@@ -1165,6 +1165,180 @@ class ContentExample {
+                  KatexSpanNode(styles: KatexSpanStyles(), text: null, nodes: [
+                    KatexSpanNode(
+                      styles: KatexSpanStyles(fontFamily: 'KaTeX_Main'),
+                      text: 'T', nodes: null),
+                    KatexSpanNode(styles: KatexSpanStyles(), text: null, nodes: []),
+                    KatexNegativeMarginNode(leftOffsetEm: -0.1667, nodes: [
+                      KatexVlistNode(rows: [
+                        KatexVlistRowNode(
+                          verticalOffsetEm: -2.7845 + 3,
+                          node: KatexSpanNode(styles: KatexSpanStyles(), text: null, nodes: [
+                            KatexSpanNode(styles: KatexSpanStyles(), text: null, nodes: [
+                              KatexSpanNode(
+                                styles: KatexSpanStyles(fontFamily: 'KaTeX_Main'),
+                                text: 'E', nodes: null),
+                            ]),
+                          ])),
+                      ]),
+                      KatexSpanNode(styles: KatexSpanStyles(), text: null, nodes: []),
+                      KatexNegativeMarginNode(leftOffsetEm: -0.125, nodes: [
+                        KatexSpanNode(
+                          styles: KatexSpanStyles(fontFamily: 'KaTeX_Main'),
+                          text: 'X', nodes: null),
+                      ]),
+                    ]),
+                  ]),
+                ]),
+              ]),
+          ]),
+        ]),
+      ]),
+    ]);
+
+  static const mathBlockKatexNegativeMarginsOnVlistRow = ContentExample(
+    'math block, KaTeX negative margins on a vlist row',
+    '```math\nX_n\n```',
+    '<p>'
+      '<span class="katex-display"><span class="katex">'
+        '<span class="katex-mathml"><math xmlns="http://www.w3.org/1998/Math/MathML" display="block"><semantics><mrow><msub><mi>X</mi><mi>n</mi></msub></mrow><annotation encoding="application/x-tex">X_n</annotation></semantics></math></span>'
+        '<span class="katex-html" aria-hidden="true">'
+          '<span class="base">'
+            '<span class="strut" style="height:0.8333em;vertical-align:-0.15em;"></span>'
+            '<span class="mord">'
+              '<span class="mord mathnormal" style="margin-right:0.07847em;">X</span>'
+              '<span class="msupsub">'
+                '<span class="vlist-t vlist-t2">'
+                  '<span class="vlist-r">'
+                    '<span class="vlist" style="height:0.1514em;">'
+                      '<span style="top:-2.55em;margin-left:-0.0785em;margin-right:0.05em;">'
+                        '<span class="pstrut" style="height:2.7em;"></span>'
+                        '<span class="sizing reset-size6 size3 mtight">'
+                          '<span class="mord mathnormal mtight">n</span></span></span></span>'
+                  '<span class="vlist-s">​</span></span>'
+                  '<span class="vlist-r">'
```

Comment:
nit:
```suggestion
                '<span class="vlist-t vlist-t2">'
                  '<span class="vlist-r">'
                    '<span class="vlist" style="height:0.1514em;">'
                      '<span style="top:-2.55em;margin-left:-0.0785em;margin-right:0.05em;">'
                        '<span class="pstrut" style="height:2.7em;"></span>'
                        '<span class="sizing reset-size6 size3 mtight">'
                          '<span class="mord mathnormal mtight">n</span></span></span></span>'
                    '<span class="vlist-s">​</span></span>'
                  '<span class="vlist-r">'
```

right?

(I.e., the .vlist-s is a child of the preceding .vlist-r, not of the .vlist-t.)